### PR TITLE
CI: add compile tests

### PR DESCRIPTION
**Pull request by iomartin**
Make sure that the project compiles.

Runs on pushes and on pull requests

### CODE REVIEW

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+on: [push, pull_request]
+
+jobs:
+   compile:
+     runs-on: ubuntu-20.04
+     steps:
+       - uses: actions/checkout@v2
+         with:
+           ref: ${{ github.event.pull_request.head.sha }}
+       - name: Install uBPF
```
Comment (sbates130272) [inline .github/workflows/compile.yml:17]: Do we want to lock this to a tag of ubpf?

File: .github/workflows/compile.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: compile test
+
+on: [push, pull_request]
```
Comment (sbates130272) [inline .github/workflows/compile.yml]: I don't think we want this on pushes to dev branches as they are often broken. I normally do this on PR against master and then also make it part of the master branch rules.

**Review by sbates130272 — CHANGES_REQUESTED** [review]

**Comment by iomartin** [comment]
@sbates130272, I fixed the code to address your comments.

I was getting some timeouts when running the CI and it turns out it was a bandwidth issue on where some of the git modules are hosted. So I cherry-picked a commit from upstream that changes the address of those modules and I added to this PR as well.

**Review by sbates130272 — APPROVED** [review]
Thanks @iomartin. Looks good!

**Pull request merged**